src/app/inventory/store/stats-conditional.ts

```diff
@@ -0,0 +1,207 @@
+    return { rule: 'adeptWeapon' };
+  }
+
+  if (enhancedIntrinsics.has(defHash)) {
+    return { rule: 'enhancedIntrinsic' };
+  }
+}
+
+/**
+ * This function indicates whether a mod's stat effect is active on the item.
+ *
+ * For example, some subclass plugs reduce a different stat per character class,
+ * which we identify using the passed subclass item, or the classType for static
+ * setups that do not include a particular item.
+ */
+export function isPlugStatActive(
+  rule: PlugStatActivityRule,
+  item: DimItem | undefined,
+  classType?: DestinyClass
+): boolean {
+  if (!rule) {
+    return true;
+  }
+
+  const warnMissingItem = () => {
+    warnLog('conditional stats', 'stat condition depends on item but we do not have an item here');
+    return true;
+  };
+  switch (rule.rule) {
+    case 'never':
+      return false;
+    case 'classType':
+      classType ??= item?.classType;
+      if (classType === undefined) {
+        warnLog(
+          'conditional stats',
+          'stat condition depends on class type but we do not have a class type here'
+        );
+        return true;
+      }
+      return classType === rule.classType;
+    case 'adeptWeapon':
+      return item ? adeptWeaponHashes.includes(item.hash) : warnMissingItem();
+    case 'masterwork':
+      return item?.masterwork ?? warnMissingItem();
+    case 'enhancedIntrinsic':
+      // Crafted weapons get bonus stats from enhanced intrinsics at Level 20+.
+      // The number 20 isn't in the definitions, so just hardcoding it here.
+      // Alternatively, enhancing an adept weapon gives it an enhanced intrinsic
+      // that gives bonus stats simply because it's an adept weapon, and more if Level 20+.
+      // stats.ts:getPlugStatValue actually takes care of scaling this to the correct bonus.
+      return item
+        ? (item.craftedInfo?.level || 0) >= 20 || adeptWeaponHashes.includes(item.hash)
+        : warnMissingItem();
+  }
+}
+
+/**
+ * Turn the investmentStats from `itemDef` into an elaborated form
+ * that more explicitly contains the plug stat activity rules and
+ * fixes some data errors/problems.
+ *
+ * TODO: If/when https://github.com/DestinyItemManager/DIM/issues/9076
+ * happens and we use DimPlug everywhere, we store the return value in
+ * DimPlug instead of caching here.
+ */
+export const mapAndFilterInvestmentStats = weakMemoize(
+  (itemDef: PluggableInventoryItemDefinition): readonly Readonly<DimPlugInvestmentStat>[] => {
+    let hasDupes: boolean | undefined;
+
+    // Fast path in case all stats are active and we need no postprocessing.
+    // This needs some knowledge of how `getPlugInvestmentStatActivityRule` works...
+    if (
+      !itemDef.traitHashes?.includes(EXOTIC_CATALYST_TRAIT) &&
+      itemDef.investmentStats.every((s) => !s.isConditionallyActive)
+    ) {
+      hasDupes =
+        uniqBy(itemDef.investmentStats, (s) => s.statTypeHash).length !==
+        itemDef.investmentStats.length;
```

Comment:
Is it worth sorting the investment stats by stat hash? Should allow for a fast duplicate scan both here and below.

Reply:
Maybe - we'd have to move the Enhanced Bipod code above this though, or make it more targeted because it'll change the indices that the fix relies on.